packages/flutter_tools/lib/src/cache.dart

```diff
@@ -597,6 +597,7 @@ abstract class CachedArtifact extends ArtifactSet {
         try {
           await cache.downloadFile(url, tempFile);
           status.stop();
+        // The exception is rethrown, so don't catch only Exceptions.
```

Comment:
Can this just be a `finally` with no `catch`?

Reply:
`status.stop()` and `status.cancel()` are different. `stop()` is for normal completion, and `cancel()` is for an abnormal completion.